code/ui/components/src/tabs/tabs.tsx

```diff
@@ -138,9 +142,13 @@ export const Tabs: FC<TabsProps> = memo(
     id: htmlId,
     menuName,
   }) => {
-    const list = useMemo<ChildrenList>(
-      () => childrenToList(children, selected),
-      [children, selected]
+    const idList = childrenToList(children).map((i) => i.id);
+    const primeList = useMemo<ChildrenList>(() => childrenToList(children), [...idList]);
+
+    const list = useMemo(
+      () =>
+        primeList.map((i, index) => ({ ...i, active: selected ? i.id === selected : index === 0 })),
+      [selected, primeList]
     );
```

Comment:
I don't think you need the `primeList` middle-step here, since that will get memoized whenever `idList` is equal, so it won't change independently of `idList`.

```suggestion
    const idList = childrenToList(children).map((i) => i.id);
    const list = useMemo(
      () =>
        childrenToList(children).map((i, index) => ({ ...i, active: selected ? i.id === selected : index === 0 })),
      [selected, ...idList]
    );
```

Reply:
I'll see if you solution works